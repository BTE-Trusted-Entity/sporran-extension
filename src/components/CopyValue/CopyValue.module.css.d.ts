export const __esModule: true;
export const line: string;
export const kiltLine: string;
export const didLine: string;
export const input: string;
export const long: string;
