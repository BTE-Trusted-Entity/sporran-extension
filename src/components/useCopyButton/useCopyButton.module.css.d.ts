declare namespace UseCopyButtonModuleCssNamespace {
  export interface IUseCopyButtonModuleCss {
    copied: string;
    copy: string;
  }
}

declare const UseCopyButtonModuleCssModule: UseCopyButtonModuleCssNamespace.IUseCopyButtonModuleCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: UseCopyButtonModuleCssNamespace.IUseCopyButtonModuleCss;
};

export = UseCopyButtonModuleCssModule;
