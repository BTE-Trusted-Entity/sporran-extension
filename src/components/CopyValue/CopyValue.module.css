.line {
  --copy-value-icon: '';

  position: relative;
  width: 100%;
  margin: 0;
}

.kiltLine {
  composes: line;

  --copy-value-icon: url('../images/kilt-coin-circle.svg');
  --copy-value-icon-size: 1rem;
}

.didLine {
  composes: line;

  --copy-value-icon: url('../images/manage-did.svg');
  --copy-value-icon-size: 1.125rem;
}

.input {
  composes: input from '../../components/Input/Input.module.css';
  font-size: 0.75rem;
  width: calc(100% - 4.5rem);
  padding-right: 2.5rem;
  padding-left: 2rem;
  background: var(--copy-value-icon) no-repeat left 0.625rem
      center/var(--copy-value-icon-size),
    var(--input-gradient);
}

.long {
  composes: input;
  font-size: calc(10.5rem / 16);
}
