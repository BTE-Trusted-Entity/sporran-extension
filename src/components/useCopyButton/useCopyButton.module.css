.copy {
  composes: buttonIcon from '../Button/Button.module.css';
  background-image: url('../images/copy.svg');
  background-position: center 40%;
  background-size: auto 50%;
  width: 2.25rem;
  height: 2.25rem;
  position: absolute;
  right: 0.25rem;
}

.copied {
  composes: copy;
  background-image: url('../images/checkmark-circle.svg');
}
