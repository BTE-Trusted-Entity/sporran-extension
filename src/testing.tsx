export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
