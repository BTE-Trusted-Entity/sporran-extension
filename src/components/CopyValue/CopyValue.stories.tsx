import { Meta } from '@storybook/react';

import { CopyValue } from './CopyValue';

export default {
  title: 'Components/CopyValue',
  component: CopyValue,
} as Meta;

export function Template(): JSX.Element {
  return <CopyValue value="FOO" label="BAR" />;
}
