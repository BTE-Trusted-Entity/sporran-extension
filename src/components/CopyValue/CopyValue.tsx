import { useRef } from 'react';

import * as styles from './CopyValue.module.css';

import { useCopyButton } from '../useCopyButton/useCopyButton';

interface Props {
  value: string;
  label?: string;
  labelledBy?: string;
  className?: string;
}

export function CopyValue({
  value,
  label,
  labelledBy,
  className = styles.line,
}: Props): JSX.Element {
  const inputRef = useRef<HTMLInputElement>(null);
  const copy = useCopyButton(inputRef);

  const longValue = value.length > 52;

  return (
    <p className={className}>
      <input
        className={longValue ? styles.long : styles.input}
        ref={inputRef}
        readOnly
        value={value}
        {...(label && { 'aria-label': label })}
        {...(labelledBy && { 'aria-labelledby': labelledBy })}
      />
      {copy.supported && (
        <button
          className={copy.className}
          onClick={copy.handleCopyClick}
          type="button"
          aria-label={copy.title}
          title={copy.title}
        />
      )}
    </p>
  );
}
