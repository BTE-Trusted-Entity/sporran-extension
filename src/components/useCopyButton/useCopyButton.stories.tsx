import { Meta } from '@storybook/react';
import { useRef } from 'react';

import { useCopyButton } from './useCopyButton';

export default {
  title: 'Components/useCopyButton',
} as Meta;

export function Template(): JSX.Element {
  const ref = useRef(null);
  const copy = useCopyButton(ref);

  return (
    <form>
      <label>
        Input: <input ref={ref} />
      </label>

      {copy.supported && (
        <button
          className={copy.className}
          type="button"
          onClick={copy.handleCopyClick}
          title={copy.title}
          aria-label={copy.title}
        />
      )}
    </form>
  );
}
