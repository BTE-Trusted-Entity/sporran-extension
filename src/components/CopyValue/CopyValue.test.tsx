import { render } from '../../testing/testing';

import '../../components/useCopyButton/useCopyButton.mock';

import { CopyValue } from './CopyValue';

describe('CopyValue', () => {
  it('should match the snapshot with label', async () => {
    const { container } = render(<CopyValue value="FOO" label="BAR" />);
    expect(container).toMatchSnapshot();
  });
  it('should match the snapshot with labelled-by', async () => {
    const { container } = render(<CopyValue value="FOO" labelledBy="ZAB" />);
    expect(container).toMatchSnapshot();
  });
});
