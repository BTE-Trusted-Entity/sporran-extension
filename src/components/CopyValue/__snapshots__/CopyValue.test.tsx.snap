// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyValue should match the snapshot with label 1`] = `
<div>
  <p
    class="line"
  >
    <input
      aria-label="BAR"
      class="input"
      readonly=""
      value="FOO"
    />
    <button
      aria-label="copy to clipboard"
      class="copy"
      title="copy to clipboard"
      type="button"
    />
  </p>
</div>
`;

exports[`CopyValue should match the snapshot with labelled-by 1`] = `
<div>
  <p
    class="line"
  >
    <input
      aria-labelledby="ZAB"
      class="input"
      readonly=""
      value="FOO"
    />
    <button
      aria-label="copy to clipboard"
      class="copy"
      title="copy to clipboard"
      type="button"
    />
  </p>
</div>
`;
